From 2841e2b61e32f26900bde987d469c8b97ea31999 Mon Sep 17 00:00:00 2001
From: Eirik Aavitsland <eirik.aavitsland@qt.io>
Date: Fri, 3 Aug 2018 13:25:15 +0200
Subject: Check for QImage allocation failure in qgifhandler

Since image files easily can be (or corrupt files claim to be) huge,
it is worth checking for out of memory situations.

Change-Id: I635a3ec6852288079fdec4e14cf7e776fe59e9e0
Reviewed-by: Lars Knoll <lars.knoll@qt.io>
---
 src/plugins/imageformats/gif/qgifhandler.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

Index: qtbase-opensource-src-5.3.2+dfsg/src/gui/image/qgifhandler.cpp
===================================================================
--- qtbase-opensource-src-5.3.2+dfsg.orig/src/gui/image/qgifhandler.cpp
+++ qtbase-opensource-src-5.3.2+dfsg/src/gui/image/qgifhandler.cpp
@@ -356,7 +356,8 @@ int QGIFFormat::decode(QImage *image, co
                     (*image) = QImage(swidth, sheight, format);
                     bpl = image->bytesPerLine();
                     bits = image->bits();
-                    memset(bits, 0, image->byteCount());
+                    if (bits)
+                        memset(bits, 0, image->byteCount());
                 }
 
                 // Check if the previous attempt to create the image failed. If it
@@ -417,6 +418,10 @@ int QGIFFormat::decode(QImage *image, co
                         backingstore = QImage(qMax(backingstore.width(), w),
                                               qMax(backingstore.height(), h),
                                               QImage::Format_RGB32);
+                        if (backingstore.isNull()) {
+                            state = Error;
+                            return -1;
+                        }
                         memset(bits, 0, image->byteCount());
                     }
                     const int dest_bpl = backingstore.bytesPerLine();
